testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt

```diff
@@ -3,111 +3,133 @@ package com.rakuten.tech.mobile.testapp.ui.settings
 import com.rakuten.tech.mobile.testapp.ui.base.BaseActivity
 import kotlinx.coroutines.launch
+import kotlin.properties.Delegates
 
-abstract class SettingsMenuBaseActivity : BaseActivity() {
+class SettingsMenuActivity : BaseActivity() {
 
     private lateinit var settings: AppSettings
 
-    private lateinit var btnSave: View
+    private lateinit var textInfo: TextView
     private lateinit var edtAppId: EditText
     private lateinit var edtSubscriptionKey: EditText
+    private lateinit var switchTestMode: SwitchMaterial
+    private lateinit var settingsProgressDialog: SettingsProgressDialog
+
+    private var saveViewEnabled by Delegates.observable(true) { _, old, new ->
+        if (new != old) {
+            invalidateOptionsMenu()
+        }
+    }
+
     private val settingTextWatcher = object: TextWatcher {
         override fun afterTextChanged(s: Editable?) {}
 
         override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
 
         override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
-            validateSetting()
+            enableSaveView()
         }
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         settings = AppSettings.instance
+        setContentView(R.layout.settings_menu_activity)
+
+        initializeActionBar()
+        initializeUIComponents()
+
+        renderAppSettingsScreen()
     }
 
     override fun onCreateOptionsMenu(menu: Menu): Boolean {
-        menuInflater.inflate(R.menu.main, menu)
+        menuInflater.inflate(R.menu.settings_menu, menu)
+        menu.findItem(R.id.settings_menu_save).isEnabled = saveViewEnabled
         return true
     }
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         return when (item.itemId) {
-            R.id.set_app_id_and_key -> showAppSettings()
+            android.R.id.home -> {
+                finish()
+                return true
+            }
+
+            R.id.settings_menu_save -> {
+                onSaveAction()
+                return true
+            }
             else -> super.onOptionsItemSelected(item)
         }
     }
 
-    private fun showAppSettings(): Boolean {
-        val settingsDialog = LayoutInflater.from(this)
-            .inflate(R.layout.app_settings_dialog, null, false)
+    private fun onSaveAction() {
+        settingsProgressDialog.show()
+
+        updateSettings(
+            edtAppId.text.toString(),
+            edtSubscriptionKey.text.toString(),
+            switchTestMode.isChecked
+        )
+    }
+
+    private fun initializeActionBar() {
+        val toolBar = findViewById<Toolbar>(R.id.toolbar)
+        setSupportActionBar(toolBar)
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.setDisplayShowHomeEnabled(true)
+    }
+
+    private fun initializeUIComponents() {
+        textInfo = findViewById(R.id.textInfo)
```

Comment:
We're using Kotlin Android Extensions, so couldn't you import these views from `kotlinx.android.synthetic.main.settings_menu_activity.*` rather than using `findViewById`?

Reply:
Yes, I have updated using synthetic imports.